assets/src/edit-story/components/panels/borderRadius/panel.js

```diff
@@ -0,0 +1,192 @@
+  bottomLeft: 0,
+  locked: true,
+};
+
+const LockToggle = styled(Toggle)`
+  position: absolute;
+  top: 50%;
+  left: calc(50% - ${TOGGLE_WIDTH / 2}px);
+`;
+
+const BoxedNumeric = styled(Numeric)`
+  padding: 6px 6px;
+  border-radius: 4px;
+`;
+
+const TOP_LEFT = 'top-left';
+const TOP_RIGHT = 'top-right';
+const BOTTOM_LEFT = 'bottom-left';
+const BOTTOM_RIGHT = 'bottom-right';
+
+const Icon = styled.div`
+  flex: 0 0 32px;
+  text-align: center;
+  padding-top: 6px;
+  ${({ corner }) => TOP_RIGHT === corner && `transform: scaleX(-1);`}
+  ${({ corner }) => BOTTOM_LEFT === corner && `transform: scaleY(-1);`}
+  ${({ corner }) => BOTTOM_RIGHT === corner && `transform: scale(-1, -1);`}
+  svg {
+    width: 24px;
+  }
+`;
+
+function BorderRadiusPanel({ selectedElements, pushUpdateForObject }) {
+  const borderRadius = useCommonObjectValue(
+    selectedElements,
+    'borderRadius',
+    DEFAULT_BORDER_RADIUS
+  );
+
+  const allSupportBorder = selectedElements.every((el) =>
+    canMaskHaveBorder(el)
+  );
+
+  const lockRadius = borderRadius.locked === true;
+
+  const handleChange = useCallback(
+    (name, value) => {
+      const newRadius = !lockRadius
+        ? {
+            [name]: value,
+          }
+        : {
+            topLeft: value,
+            topRight: value,
+            bottomRight: value,
+            bottomLeft: value,
+          };
+      pushUpdateForObject(
+        'borderRadius',
+        newRadius,
+        DEFAULT_BORDER_RADIUS,
+        true
+      );
+    },
+    [pushUpdateForObject, lockRadius]
+  );
+
+  const handleLockChange = useCallback(
+    (locked) => {
+      const newRadius = locked
+        ? {
+            locked,
+            topLeft: borderRadius.topLeft,
+            topRight: borderRadius.topLeft,
+            bottomRight: borderRadius.topLeft,
+            bottomLeft: borderRadius.topLeft,
+          }
+        : {
+            locked,
+          };
+      pushUpdateForObject(
+        'borderRadius',
+        newRadius,
+        DEFAULT_BORDER_RADIUS,
+        true
+      );
+    },
+    [pushUpdateForObject, borderRadius]
+  );
+
+  if (!allSupportBorder) {
+    return null;
+  }
+
+  return (
+    <SimplePanel name="borderRadius" title={__('Corner radius', 'web-stories')}>
+      <Row>
+        <BoxedNumeric
+          value={borderRadius.topLeft}
+          aria-label={__('Edit: Top left corner radius', 'web-stories')}
+          onChange={(value) => handleChange('topLeft', value)}
+        />
+        {/* Default, passed for readability only */}
+        <Icon corner={TOP_LEFT}>
+          <BorderLockLine />
+        </Icon>
+        <Icon corner={TOP_RIGHT}>
+          <BorderLockLine />
+        </Icon>
+        <LockToggle
+          icon={<Lock />}
+          uncheckedIcon={<Unlock />}
+          value={borderRadius.locked}
+          onChange={() => handleLockChange(!borderRadius.locked)}
+          aria-label={__('Toggle corner radius lock', 'web-stories')}
+        />
```

Comment:
Is it by design, that the lock comes between the top-left and top-right input? Is that the right position in the tab-order? It might be, but I feel like it's better to have it between the top and bottom corners. I'd add a row in-between the two other rows with negative margins top and bottom and just the toggle centered.

Unless we want the lock to be between the two top corners in the tab-order - then it's probably best as is, though one could argue it would fit better _between_ the two icons in the DOM, but that's a tiny detail.

Reply:
Hmm, it's not by design necessarily, TBH then when tabbing through it I was also surprised a few times that it jumped to the lock right away -- will change as suggested.